applications/nrf_desktop/src/hw_interface/Kconfig.board

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2019-2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config DESKTOP_BOARD_DEF_PATH
+	string "Configuration file"
+	default "port_state_def.h"
```

Comment:
Still not right approach.

Looking at this file:
https://github.com/nrfconnect/sdk-nrf/blob/b4d8da7d266ad1d511ed9554c8c153c062f9ae74/applications/nrf_desktop/configuration/nrf52810dmouse_nrf52810/port_state_def.h#L30-L36

then I assume what you would want to configure `name`  for the port in use, like `gpio0`, right ?

That does sound like something where there should probably have been a `chosen` instead.
Maybe take a look at https://github.com/nrfconnect/sdk-nrf/pull/5439 to see how `chosen` node was used for handling external flash support.